.github/workflows/test.yml

```diff
@@ -1,14 +1,25 @@
 name: test
-on: [ push, pull_request ]
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
What does this do?

Reply:
This stops your previous workflow when your HEAD got updated.

Reply:
got it, cool!  I'm very unfamiliar with github actions still :) (more of a gitlab guy)